src/BlockchainData.js

```diff
@@ -0,0 +1,228 @@
+import React, { Component } from 'react';
+import Web3 from 'web3';
+import Tx from 'ethereumjs-tx';
+import EthCrypto from 'eth-crypto';
+import { MULTISIG_ABI, MULTISIG_ADDRESS, MULTISIG_ERC20_ABI, MULTISIG_ERC20_ADDRESS, ERC20_ABI, 
+	TOKENIMPERIAL_ADDRESS, TOKENDEMOCRATIC_ADDRESS, TOKENIMPERIAL_SYMBOL, TOKENDEMOCRATIC_SYMBOL } from './config';
+import SmartContract from './SmartContract'; 
+import ERC20 from './erc20/ERC20';
+import './css/ButtonGroup.css';
+
+
+/* Validate forms, clear data in forms, check that transaction transfer works correctly*/ 
+class BlockchainData extends Component {
+
+	constructor(props) {
+		super(props);
+		this.state = {
+			web3: '',
+			multisig: '',
+			etherAddress: '',
+			etherBalance: '',
+			tokenImperialBalance: '',
+			tokenDemocraticBalance: '',
+			contractBalance: '',
+			tokenImperialContractBalance: '',
+			tokenDemocraticContractBalance: '',
+			Ethereum: true
+		}
+	}
+	static getDerivedStateFromProps(props, state) {
+    	return {etherAddress: props.etherAddress };
+  	}
+
+  	componentDidMount() {
+  		const web3 = new Web3('https://ropsten.infura.io/v3/a33baa265ae340c29c82373e91533edf');
+  		this.setState({ web3: web3 });
+  		const multisig = new web3.eth.Contract(MULTISIG_ABI, MULTISIG_ADDRESS);
+		this.setState({ multisig: multisig });
```

Comment:
it is bad practice to setState multiple times, better to form state object and do setState once, so it won't cause extra rerenderings.